### ISSUE TITLE
Vagrant assuming OS specific paths, breaks under Cygwin

### ISSUE THREAD

**Issue by dragon788** [mitchellh/vagrant#5747]
Error below:
```
{ US114469 }  » vagrant box add trusty64 ~/Downloads/trusty64-0.0.93.box
C:/HashiCorp/Vagrant/embedded/lib/ruby/2.0.0/uri/common.rb:176:in `split': bad URI(is not URI?): file:C:\cygdrive\c\Users\es29866\Downloads\trusty64-0.0.93.box (URI::InvalidURIError)
from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.0.0/uri/common.rb:211:in `parse'
from C:/HashiCorp/Vagrant/embedded/lib/ruby/2.0.0/uri/common.rb:747:in `parse'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/box_add.rb:454:in `metadata_url?'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/box_add.rb:77:in `block in call'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/box_add.rb:75:in `map'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builtin/box_add.rb:75:in `call'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/warden.rb:34:in `call'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/builder.rb:116:in `call'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `block in run'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/util/busy.rb:19:in `busy'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/action/runner.rb:66:in `run'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/plugins/commands/box/command/add.rb:85:in `execute'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/plugins/commands/box/command/root.rb:61:in `execute'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/cli.rb:42:in `execute'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/lib/vagrant/environment.rb:301:in `cli'
from C:/HashiCorp/Vagrant/embedded/gems/gems/vagrant-1.7.2/bin/vagrant:174:in `<main>'
```    

I use babun, a preconfigured Cygwin environment to keep a friendly and familiar bash/zsh shell on Windows, and do most of my development under Linux VMs. When I try to add a box using the ~/path syntax, because Cygwin has to do some redirection to use Windows style paths, it breaks. I think this is demonstrated pretty well in the error block above. Let me know if you need more information.

**Comment by sethvargo** [comment]
Hi @dragon788 

I am sorry you are having issues with Vagrant. Can you please share your complete `Vagrantfile` as well as the output of the command in [debug mode](http://docs.vagrantup.com/v2/other/debugging.html)? Thanks!

**Comment by dragon788** [comment]
Hi Seth, this is actually pre-Vagrantfile when trying to add a box that was created via Packer. The issue is when doing the ```vagrant box add trusty64 path/to/my/box``` that Vagrant is assuming some OS specific paths, not using an agnostic pathing library to resolve. I was able to successfully add the box if I changed directory into the Downloads folder before running the add, so it didn't have to navigate any non-native paths. I have a feeling the ~ which is used in Cygwin but not Windows threw it off.
